dev_tools/requirements/envs/dev.env.txt

```diff
@@ -1,5 +1,5 @@
 #
-# This file is autogenerated by pip-compile with Python 3.10
+# This file is autogenerated by pip-compile with Python 3.12
```

Comment:
since the pinned requirements are very delicate w.r.t. exactly which python environment you run them from, we use the `..from_docker.sh` script to bump the dependencies from the included docker image. As an example, you can see that your name is now forever immortalized in the requirements comments. 

I'll run the script and push to this pr to save you some time